File: Back/Domain/Discount.cs
using Domain.Enums;

namespace Domain;
public class Discount
{
    [Key]
    public int Id { get; set; }
    public required DiscountType Type { get; set; }
    public required string ProductId { get; set; }
    public string SourceProductId { get; set; }
    public int SourceRequiredQuantity { get; set; }
    public decimal Percentage { get; set; }
    public DateTime StartDate { get; set; }
    public DateTime EndDate { get; set; }
    public bool IsActive { get; set; }

}

File: Front/TheShop/src/app/models/discount.ts

export class Discount {
  id: number;
  type: string;
  productId: string;
  sourceProductId?: string;
  sourceRequiredQuantity: number;
  percentage: number;
  startDate: Date;
  endDate: Date;

  constructor(
    id: number,
    type: string,
    productId: string,
    sourceProductId: string | undefined,
    sourceRequiredQuantity: number,
    percentage: number,
    startDate: Date,
    endDate: Date
  ) {
    this.id = id;
    this.type = type;
    this.productId = productId;
    this.sourceProductId = sourceProductId;
    this.sourceRequiredQuantity = sourceRequiredQuantity;
    this.percentage = percentage;
    this.startDate = startDate;
    this.endDate = endDate;
  }
}
